ui/src/containers/HeaderContainer.js

```diff
@@ -56,8 +56,8 @@ HeaderContainer.propTypes = {
 
 function mapStateToProps (state) {
   return {
-    currentUser: state.header.user,
-    isUserRequestFailed: state.header.isUserRequestFailed
+    currentUser: state.user.user,
```

Comment:
```suggestion
    currentUser: state.user.username,
```
Should this be user.username?

Reply:
`user` is the name of the reducer. the second `user` is the user object, containing the email and username. It is a bit confusing, I considered flattening the `user` reducer so that they are all on the same level but thought it better to keep the actual user as a lone object, so it can be imported into components without everything else that is irrelevant.